version/src/main/java/io/smallrye/common/version/VersionRange.java

```diff
@@ -0,0 +1,113 @@
+package io.smallrye.common.version;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.function.Predicate;
+
+/**
+ * A {@link VersionRange} is a predicate that tests if a version string is within a specified range.
+ */
+public class VersionRange implements Predicate<String> {
+
+    private static final Map<String, VersionRange> CACHE_SPEC = Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
I wouldn't bother with a cache, to be honest.

Reply:
Gotcha, I reverted in a separate commit now